### PR TITLE
fix: docs

### PR DESCRIPTION
**Pull request by yuwen01**
1. Add more docs and comments to all files.
2. Refactor group compression / serialization methods to a utils file.
3. Refactor vkey hash to take a string as input, since that's what `vk.bytes32()` outputs.
4. Remove the unsupported v1.2.0 vkey and add the new v3.0.0rc4 groth16 vkey

### CODE REVIEW

File: example/script/src/main.rs [hunk 1]
```diff
@@ -62,7 +64,9 @@ async fn main() {
     if args.prove {
         // Initialize the prover client
         let client = ProverClient::new();
-        let (pk, _vk) = client.setup(ELF);
+        let (pk, vk) = client.setup(ELF);
+
+        println!("vkey bytes {:?}", sp1_sdk::HashableKey::bytes32(&vk));
```
Comment (ratankaliani) [inline example/script/src/main.rs]: nit: "Verification Key: "

File: verifier/src/lib.rs [hunk 2]
```diff
@@ -1,255 +1,39 @@
-use ark_bn254::{Fq, G1Affine};
-use ark_ff::PrimeField;
-use ark_serialize::CanonicalSerialize;
-use borsh::BorshSerialize;
+//! # Verifier
+//!
+//! This crate contains utilities for verifying SP1 Groth16 proofs on Solana.
+//!
+//! # Example
+//! ```
```
Comment (ratankaliani) [inline verifier/src/lib.rs:6]: Does this need to have the tag like ```rust on it?

**Review by ratankaliani — APPROVED** [review]
Some small comments, but looks way cleaner! LGTM

Can you add a PR description?

File: verifier/src/lib.rs (hunk 2 above)
Comment (yuwen01) [reply verifier/src/lib.rs:6]: to run as a doctest, no, since it's in a rust file

**Pull request merged**